File: src/test/baekjoon/java/bruteforce/Bunhaehab.java
package test.baekjoon.java.bruteforce;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Bunhaehab {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());

        int answer = 0;
        int start = N - (String.valueOf(N).length() * 9);
        for(int i = start; i < N; i++) {
            int num = i, sum = i;
            while(num != 0) {
                sum += num % 10;
                num /= 10;
            }

            if(N == sum) {
                answer = i;
                break;
            }
        }

        System.out.println(answer);
    }
}

File: src/test/baekjoon/python/bruteforce/Bunhaehab.py
N = int(input())
start = N - (len(str(N)) * 9)
answer = 0
for i in range(start, N):
    if i < 1:
        continue

    _sum = num = i
    while num != 0:
        _sum += num % 10
        num //= 10
    if _sum == N:
        answer = i
        break
print(answer)

